src/lib/components/Callback/Callback.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Actions, SupAuth, history } from "gisida";
+
+const { defaultUnSupAuthZ: deAuthZ } = SupAuth;
+
+class Callback extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { loaded: false };
+    this.history = history;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (
+      !this.state.loaded &&
+      nextProps.global.APP &&
+      nextProps.global.APP.loaded
+    ) {
+      this.setState(
+        {
+          loaded: true,
+          APP: { ...nextProps.global.APP }
+        },
+        () => {
+          this.authorizeUser(this.state.APP);
+        }
+      );
+    }
+  }
+
+  async authorizeUser(APP) {
+    const { dispatch } = this.props;
+    const { AUTH } = this.props.global;
+    const accessToken = this.getAccessToken();
+    const expiryTime = this.getExpiryTime();
+    localStorage.setItem("expiry_time", expiryTime);
+    dispatch(Actions.receiveToken(accessToken));
+    const { isAuth, authConfig, user } = await SupAuth.authorizeUser(
+      APP,
+      AUTH,
+      accessToken
+    );
+    if (isAuth && authConfig) {
+      dispatch(Actions.getAuthConfigs(authConfig));
+    }
```

Comment:
Should we only be setting the `expiry_time` after we confirm the user is authenticated? Otherwise shouldn't we unset the `expiry_time` if the user doesn't fully authenticate?

Reply:
> Should we only be setting the `expiry_time` after we confirm the user is authenticated? Otherwise shouldn't we unset the `expiry_time` if the user doesn't fully authenticate?

This is has been updated.